playwright/tests/desk-tool/documentActions.spec.ts

```diff
@@ -0,0 +1,113 @@
+import {test, expect} from '@playwright/test'
+import {createUniqueDocument, DEFAULT_DATASET, deleteDocument} from '../../helpers'
+
+/* 
+  **NOTE**
+  In these tests, we are not testing input events, so to speed up the tests we will use use `page.fill()` instead of `page.type()` to populate the document title field 
+*/
+
+const DOC_TITLE = 'Document actions [Playwright]'
+const inputSelector = 'data-testid=input-title >> input'
+
+// The path of the network request we want to wait for
+const documentActionPath = (action) =>
+  `*/**/mutate/${DEFAULT_DATASET}?tag=sanity.studio.document.${action}*`
+
+// Helper to generate path to document created in a test
+const generateDocumentUri = (docId) =>
+  `/${DEFAULT_DATASET}/desk/input-ci;documentActionsCI;${docId}`
+
+// Helper to get the path to the document the client creates for a test
+async function getDocumentUri(_type: string, title?: string) {
+  const testDoc = await createUniqueDocument({_type, title})
+  return [generateDocumentUri(testDoc?._id), testDoc?._id]
+}
+
+test.describe('@sanity/desk-tool: document actions', () => {
+  // Before we run our tests, delete all previously created test documents to avoid clutter
+  test.beforeAll(async () => {
+    await deleteDocument({query: `*[_type == "documentActionsCI"]`})
+  })
```

Comment:
I feel like this is the wrong place to be doing a cleanup, and that we should rather clean up only this current test run when complete. What do you think? Could we change this to a `afterAll` with a more narrowly scoped query?

psuedo-code:

```suggestion
  // After the tests have been run, try to clean up the documents we created in this run, along with any documents of this type older than the defined threshold (to account for cancelled/failed tests)
  test.afterAll(async () => {
    const threshold = new Date(Date.now() - STALE_TEST_THRESHOLD_MS).toISOString()
    await deleteDocument({query: `*[_type == "documentActionsCI" && (runId == "${TEST_RUN_ID}" || _createdAt < "${threshold}")]`})
  })
```

Reply:
Yeah I thought about that, but isn't it nice to see the documents that were created after you've run your tests? In case you need to debug something 🤔

Reply:
Potentially, but you can't do that with the current code if someone (or some CI) runs the test suite before you have time to debug, as it will delete all documents of that type.

We need some sort of isolation. I guess the question is whether or not we should clean up documents at all, and if so, in what way. Maybe the [`testInfo.errors` property in `afterAll`](https://playwright.dev/docs/api/class-test#test-after-all) contains whether or not the suite failed? If so, maybe we could use that to keep failed test runs around, or something? 

If not we should probably just remove the cleanup for now.

Reply:
Yeah we can use `testInfo` to determine whether we should delete a document or not, will add! Great idea 👍